src/znc.cpp

```diff
@@ -1098,42 +1098,12 @@ bool CZNC::LoadGlobal(CConfig& config, CString& sError) {
 		m_bProtectWebSessions = sVal.ToBool();
 	if (config.FindStringEntry("hideversion", sVal))
 		m_bHideVersion = sVal.ToBool();
-
-	if (config.FindStringEntry("sslprotocols", m_sSSLProtocols)) {
-		VCString vsProtocols;
-		m_sSSLProtocols.Split(" ", vsProtocols, false, "", "", true, true);
-
-		for (CString& sProtocol : vsProtocols) {
-			unsigned int uFlag = 0;
-			bool bEnable = sProtocol.TrimPrefix("+");
-			bool bDisable = sProtocol.TrimPrefix("-");
-
-			if (sProtocol.Equals("All")) {
-				uFlag = ~0;
-			} else if (sProtocol.Equals("SSLv2")) {
-				uFlag = Csock::EDP_SSLv2;
-			} else if (sProtocol.Equals("SSLv3")) {
-				uFlag = Csock::EDP_SSLv3;
-			} else if (sProtocol.Equals("TLSv1")) {
-				uFlag = Csock::EDP_TLSv1;
-			} else if (sProtocol.Equals("TLSv1.1")) {
-				uFlag = Csock::EDP_TLSv1_1;
-			} else if (sProtocol.Equals("TLSv1.2")) {
-				uFlag = Csock::EDP_TLSv1_2;
-			} else {
-				CUtils::PrintError("Invalid SSLProtocols value [" + sProtocol + "]");
-				CUtils::PrintError("The syntax is [SSLProtocols = [+|-]<protocol> ...]");
-				CUtils::PrintError("Available protocols are [SSLv2, SSLv3, TLSv1, TLSv1.1, TLSv1.2]");
-				return false;
-			}
-
-			if (bEnable) {
-				m_uDisabledSSLProtocols &= ~uFlag;
-			} else if (bDisable) {
-				m_uDisabledSSLProtocols |= uFlag;
-			} else {
-				m_uDisabledSSLProtocols = ~uFlag;
-			}
+	if (config.FindStringEntry("sslprotocols", sVal)) {
+		if (!SetSSLProtocols(sVal)) {
+			CUtils::PrintError("Invalid SSLProtocols value.");
+			CUtils::PrintError("The syntax is [SSLProtocols = [+|-]<protocol> ...]");
+			CUtils::PrintError("Available protocols are [SSLv2, SSLv3, TLSv1, TLSv1.1, TLSv1.2]");
```

Comment:
LGTM, except that now this list is detached from implementation, which makes it more likely to forget to update, whenever TLS1.3 comes.